### PR TITLE
Context broker checker

### PR DESCRIPTION
**Pull request by chicco785**
This pull request add to the validator the possibility to check if JSON examples are supported by Orion Context Broker

### CODE REVIEW

File: validator/libs/checks.js [hunk 1]
```diff
@@ -159,6 +172,66 @@ module.exports = {
     return check;
   },
 
+  //check if an example is supported by contextBroker
+  exampleSupported: function(fullPath) {
+    var check = true;
+    var apiInstance=getApiEntityClient();
```
Comment (jmcanterafonseca) [inline validator/libs/checks.js]: use linter

File: validator/libs/checks.js [hunk 2]
```diff
@@ -159,6 +172,66 @@ module.exports = {
...
+
+    var opts = {
+      options: "keyValues"
+    };
+
+    var files = schema.getFiles(fullPath + path.sep + 'example*.json');
+
+    try {
+      files.forEach(function(fileName) {
+        var body = schema.openFile(fileName, 'example ' + fileName);
```
Comment (jmcanterafonseca) [inline validator/libs/checks.js]: linter

File: validator/libs/checks.js [hunk 3]
```diff
@@ -159,6 +172,66 @@ module.exports = {
...
+
+        var createEntity = deasync(function (body, cb){
```
Comment (jmcanterafonseca) [inline validator/libs/checks.js]: linter

File: validator/libs/checks.js [hunk 4]
```diff
@@ -159,6 +172,66 @@ module.exports = {
...
+              cb(null, data);
+            }
+          });
+        });
+
+        createEntity(body);
+
+        var entityId = body.id; // String | Id of the entity to be deleted
+
+        var deleteEntity = deasync(function (entityId, cb){
```
Comment (jmcanterafonseca) [inline validator/libs/checks.js]: linter

File: validator/README.md [hunk 5]
```diff
@@ -77,6 +81,6 @@ Options available are:
 * `ajv:allErrors`: when `true` check all rules collecting all errors, when `false` return after the first error. See [ajv](https://github.com/epoberezkin/ajv) for more details. Default value: `true`
 
 ## Compiling the validator
-Should you want to change the code of the validator, to install your modified version, you  simple need compile it with the following command:
+Should you want to change the code of the validator, to install your modified version, you simply need compile it with the following command:
```
Comment (jmcanterafonseca) [inline validator/README.md]: need to

File: validator/libs/checks.js [hunk 6]
```diff
@@ -159,6 +172,66 @@ module.exports = {
...
+      files.forEach(function(fileName) {
+        var body = schema.openFile(fileName, 'example ' + fileName);
+
+        var createEntity = deasync(function (body, cb){
+          apiInstance.createEntity(body, opts, function(error, data, response) {
+            if (error){
+              check = false;
+              cb(error, null);
+            } else {
+              msg.addSupportedExample(fullPath, fileName + ' is supported');
```
Comment (jmcanterafonseca) [inline validator/libs/checks.js]: indentation

File: validator/libs/checks.js [hunk 7]
```diff
@@ -159,6 +172,66 @@ module.exports = {
...
+        deleteEntity(entityId);
+      });
+    } catch (err) {
+       msg.addError(fullPath, 'JSON Example is not supported by ' +
+         'contextBroker: ' +  JSON.stringify(err));
+       if (conf.failErrors)
+         throw new Error('Fail on Error: JSON Example is not supported by ' +
+         'contextBroker: ' +  JSON.stringify(err));
+    }
+    debug("*exampleSupported* - " + fullPath + ": "+ check);
```
Comment (jmcanterafonseca) [inline validator/libs/checks.js]: linter

File: validator/libs/checks.js [hunk 8]
```diff
@@ -159,6 +172,66 @@ module.exports = {
...
+    var check = true;
+    var apiInstance=getApiEntityClient();
+
+    var opts = {
+      options: "keyValues"
+    };
+
+    var files = schema.getFiles(fullPath + path.sep + 'example*.json');
+
+    try {
```
Comment (jmcanterafonseca) [inline validator/libs/checks.js]: linter, indentation

**Comment by jmcanterafonseca** [comment]
@chicco785 build is failing

**Comment by chicco785** [comment]
The build failure is due to the missing encoding of the [sensor value](https://travis-ci.org/Fiware/dataModels/jobs/320131008#L565) (as by issue #214), if you accept the PR #216, i will then rebase and the build should pass.

**Comment by chicco785** [comment]
@jmcanterafonseca build is now passing

**Pull request merged**